hedera-node/src/main/java/com/hedera/services/store/contracts/HederaAbstractWorldUpdater.java

```diff
@@ -68,20 +84,23 @@ protected HederaUpdateTrackingAccount track(final HederaUpdateTrackingAccount ac
 
 	@Override
 	public EvmAccount createAccount(final Address address, final long nonce, final Wei balance) {
```

Comment:
I think that we will need different logic for this one. We will need the sponsor and that would break the interface. Our custom Hedera Create opcode will have to pass that info here. That means that we will need to add that "custom" create logic in the `HederaWorldUpdater`

Reply:
Another way to implement it would be to use the updater to pass that info to the world state HederaWorldState.addPropertiesFor(Address address, String memo, JKey key, Id proxyAccount) 
and in HederaWorldState.persist() inject the props to the Account object before accountsStore.persist